pkg/operator/webhooksupportabilitycontroller/degraded_webhook.go

```diff
@@ -0,0 +1,104 @@
+package webhooksupportabilitycontroller
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"sort"
+	"strings"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// webhookInfo generically represents a webhook
+type webhookInfo struct {
+	Name     string
+	Service  *serviceReference
+	CABundle []byte
+}
+
+// serviceReference generically represents a service reference
+type serviceReference struct {
+	Namespace string
+	Name      string
+	Port      *int32
+}
+
+// updateWebhookConfigurationDegraded updates the condition specified after
+// checking that the services associated with the specified webhooks exist
+// and have at least one ready endpoint.
+func (c *webhookSupportabilityController) updateWebhookConfigurationDegraded(ctx context.Context, condition operatorv1.OperatorCondition, webhookInfos []webhookInfo) v1helpers.UpdateStatusFunc {
+	var serviceMsgs []string
+	var tlsMsgs []string
+	for _, webhook := range webhookInfos {
+		if webhook.Service != nil {
+			err := c.assertService(webhook.Service)
```

Comment:
is this necessary? since we will hit the service network anyway.

Reply:
This gives us a chance to give them a higher level "service does not exist" message vs. just saying "can't connect".